Sources/CryptoBoringWrapper/EC/EllipticCurvePoint.swift

```diff
@@ -12,41 +12,262 @@
 //
 //===----------------------------------------------------------------------===//
 @_implementationOnly import CCryptoBoringSSL
+@_implementationOnly import CCryptoBoringSSLShims
+import struct Foundation.Data
+import protocol Foundation.ContiguousBytes
 
 /// A wrapper around BoringSSL's EC_POINT with some lifetime management.
 @usableFromInline
-package class EllipticCurvePoint {
+package final class EllipticCurvePoint {
     /* private but @usableFromInline */ @usableFromInline var _basePoint: OpaquePointer
 
     @usableFromInline
-    package init(multiplying scalar: ArbitraryPrecisionInteger, on group: BoringSSLEllipticCurveGroup) throws {
+    package init(copying pointer: OpaquePointer, on group: BoringSSLEllipticCurveGroup) throws {
         self._basePoint = try group.withUnsafeGroupPointer { groupPtr in
-            guard let basePoint = CCryptoBoringSSL_EC_POINT_new(groupPtr) else {
+            guard let pointPtr = CCryptoBoringSSL_EC_POINT_dup(pointer, groupPtr) else {
                 throw CryptoBoringWrapperError.internalBoringSSLError()
             }
-            return basePoint
+            return pointPtr
+        }
+    }
+
+    @usableFromInline
+    package convenience init(copying other: EllipticCurvePoint, on group: BoringSSLEllipticCurveGroup) throws {
+        try self.init(copying: other._basePoint, on: group)
+    }
+
+    @usableFromInline
+    package init(multiplying scalar: ArbitraryPrecisionInteger, on group: BoringSSLEllipticCurveGroup) throws {
+        self._basePoint = try scalar.withUnsafeBignumPointer { scalarPtr in
+            try group.withUnsafeGroupPointer { groupPtr in
+                guard
+                    let pointPtr = CCryptoBoringSSL_EC_POINT_new(groupPtr),
+                    CCryptoBoringSSL_EC_POINT_mul(groupPtr, pointPtr, scalarPtr, nil, nil, nil) == 1
```

Comment:
Not a new issue, but this leaks the `pointPtr` if the mul fails.

Reply:
I've addressed this by factoring out the `EC_POINT_new` into a designated `init(_pointAtInfinityOn group:)` and made all these `convenience` initialisers.